keycast.el

```diff
@@ -216,7 +220,14 @@ instead."
   (when keycast-log-mode
     (keycast-log-update-buffer))
   (when keycast-mode
-    (force-mode-line-update)))
+    ;; Mode line isn't updated for minibuffers, so try to do so from
+    ;; the window that activated it, falling back to no buffer change.
+    (if-let ((win (and (minibufferp)
+                       (minibuffer-selected-window)))
+             (buf (window-buffer win)))
+        (with-current-buffer buf
+          (force-mode-line-update))
+      (force-mode-line-update))))
```

Comment:
Note that this could also be written (a) like so:

```el
(if (minibufferp)
    (when-let ((win (minibuffer-selected-window))
               (buf (window-buffer win)))
      (with-current-buffer buf
        (force-mode-line-update)))
  (force-mode-line-update))
```
The problem with this being there is no fallback (I don't know whether `minibuffer-selected-window` and `window-buffer` are guaranteed to succeed).

Or (b) like so:

```el
(if-let (((minibufferp))
         (win (minibuffer-selected-window))
         (buf (window-buffer win)))
    (with-current-buffer buf
      (force-mode-line-update))
  (force-mode-line-update))
```

The problem with this being the syntax is not supported in Emacs 25.

Feel free to refactor, force-push, or reimplement as you see best.